src/inheritAttributes.js

```diff
@@ -1,37 +1,37 @@
-import { flatten, from } from './utils/list';
+import { flatten } from './utils/list';
 import { shallowMerge, getAttributes } from './utils/object';
 import { parseDuration } from './utils/time';
+import { findChildren, getContent } from './utils/xml';
 import resolveUrl from './resolveUrl';
 import errors from './errors';
 
 export const rep = mpdAttributes => (period, periodIndex) => {
-  const adaptationSets = from(period.getElementsByTagName('AdaptationSet'));
+  const adaptationSets = findChildren(period, 'AdaptationSet');
 
   const representationsByAdaptationSet = adaptationSets.map(adaptationSet => {
     const adaptationSetAttributes = getAttributes(adaptationSet);
 
-    const role = adaptationSet.getElementsByTagName('Role')[0];
+    const role = findChildren(adaptationSet, 'Role')[0];
     const roleAttributes = { role: getAttributes(role) };
 
     const attrs = shallowMerge({ periodIndex }, mpdAttributes, adaptationSetAttributes, roleAttributes);
 
-    const segmentTemplate = adaptationSet.getElementsByTagName('SegmentTemplate')[0];
-    const segmentList = adaptationSet.getElementsByTagName('SegmentList')[0];
-    const segmentBase = adaptationSet.getElementsByTagName('SegmentBase')[0];
+    const segmentTemplate = findChildren(adaptationSet, 'SegmentTemplate')[0];
+    const segmentList = findChildren(adaptationSet, 'SegmentList')[0];
+    const segmentBase = findChildren(adaptationSet, 'SegmentBase')[0];
 
     const segmentType = {
       segmentTemplate: segmentTemplate && getAttributes(segmentTemplate),
       segmentList: segmentList && getAttributes(segmentList),
       segmentBase: segmentBase && getAttributes(segmentBase)
     };
 
-    const representations = from(adaptationSet.getElementsByTagName('Representation'));
+    const representations = findChildren(adaptationSet, 'Representation');
 
     const inherit = representation => {
       // vtt tracks may use single file in BaseURL
-      let baseUrl = representation.getElementsByTagName('BaseURL')[0];
-
-      baseUrl = baseUrl && baseUrl.innerHTML || '';
+      const BaseUrl = findChildren(representation, 'BaseURL')[0];
```

Comment:
Minor, but it's odd using two variables with the same name, the only difference being the capitalization. If two variables are needed, they reflect different things, so we could label them differently. For instance, maybe `baseUrlElement` for the first one.

Reply:
I'll rename it, the variable represents the `BaseURL` xml node so I'll describe it as such